### PR TITLE
File type errors in uploads

### LINKED ISSUE #550: Need error for when user tries to upload non-ipynb file
When I drag and drop a file that does not have a ipynb extension, I don't get an error message. I expect an error telling me that I can only upload that file type.

### PR DESCRIPTION
**Pull request by katevoss**
Fixes #550 

Adds 3 errors for when a user uploads a file that is:
1. not .ipynb to a notebook
2. not .json to a tool
3. invalid json

### CODE REVIEW

**Review by MattPutnam — APPROVED** [review]
👍

File: src/pages/workspaces/workspace/Notebooks.js [hunk 1]
```diff
@@ -249,6 +249,7 @@ class NotebooksContent extends Component {
       acceptStyle: { cursor: 'copy' },
       rejectStyle: { cursor: 'no-drop' },
       ref: this.uploader,
+      onDropRejected: () => reportError('Error uploading notebook', 'The selected file is not a ipynb notebook file. To import a notebook, upload a file with a .ipynb extension.'),
```
Comment (zarsky-broad) [inline src/pages/workspaces/workspace/Notebooks.js]: Maybe make the short text 'Not a valid notebook'?

File: src/pages/workspaces/workspace/tools/WorkflowView.js [hunk 2]
```diff
@@ -321,6 +325,7 @@ class WorkflowViewContent extends Component {
       style: { padding: `1rem ${sideMargin} 0`, flex: 1, minHeight: 500 },
       activeStyle: { backgroundColor: colors.blue[3], cursor: 'copy' },
       ref: this.uploader,
+      onDropRejected: () => reportError('Error uploading inputs', 'The selected file is not a json file. To import inputs for this tool, upload a file with a .json extension.'),
```
Comment (zarsky-broad) [inline src/pages/workspaces/workspace/tools/WorkflowView.js]: Maybe make the short text 'Only JSON files can be uploaded' or something?

**Review by zarsky-broad — APPROVED** [review]

**Review by panentheos — APPROVED** [review]
Looks good.

File: src/pages/workspaces/workspace/Notebooks.js (hunk 1 above)
Comment (katevoss) [reply src/pages/workspaces/workspace/Notebooks.js]: Do you mean to make the error `reportError('Not a valid notebook', 'The selected file is not a ipynb notebook file. To import a notebook, upload a file with a .ipynb extension.')` ?

Error heading:
> **Not a valid notebook**

click on details shows:
> The selected file is not a ipynb notebook file. To import a notebook, upload a file with a .ipynb extension.

**Comment by zarsky-broad** [comment]
Yup.
-- 
(he/him)

**Review by elizzhao — APPROVED** [review]

File: src/pages/workspaces/workspace/tools/WorkflowView.js (hunk 2 above)
Comment (katevoss) [reply src/pages/workspaces/workspace/tools/WorkflowView.js]: How about the short text as `Not valid json`: `The selected file is not a json file. To import inputs for this tool, upload a file with a .json extension.` ?

**Pull request merged**